src/EprPrnIntegration.Common/Service/EmailService.cs

```diff
@@ -126,21 +122,16 @@ public void SendIssuedPrnsReconciliationEmailToNpwd(DateTime reportDate, int rep
 
     public void SendUpdatedPrnsReconciliationEmailToNpwd(DateTime reportDate, string reportCsv, int rowCount)
     {
+        ArgumentException.ThrowIfNullOrWhiteSpace(reportCsv, nameof(reportCsv));
+
         var templateId = _messagingConfig.NpwdReconcileUpdatedPrnsTemplateId;
-        var filename = $"reconciledprns_{reportDate:yyyyMMdd}.csv";
         var emailAddress = _messagingConfig.NpwdEmail;
 
         var messagePersonalisation = new Dictionary<string, object>
         {
-            {
-                "date", reportDate.ToString("dd/MM/yyyy")
-            },
-            {
-                "link_to_file", NotificationClient.PrepareUpload(System.Text.Encoding.UTF8.GetBytes(reportCsv), filename)
-            },
-            {
-                "row_count", rowCount
-            }
+            ["date"] = reportDate.ToString("dd/MM/yyyy"),
+            ["link_to_file"] = reportCsv,
+            ["row_count"] = rowCount
```

Comment:
Since now we are doing changes on Template as well, Can we change link_to_file to something like csvData inline with what we are doing.